Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerSortFilterProxyModel.cpp

```diff
@@ -34,7 +34,24 @@ namespace AzToolsFramework
 
     bool EntityOutlinerSortFilterProxyModel::lessThan(const QModelIndex& leftIndex, const QModelIndex& rightIndex) const
     {
-        return sourceModel()->data(leftIndex).toString() < sourceModel()->data(rightIndex).toString();
+        if (leftIndex.isValid() && rightIndex.isValid())
+        {
+            QVariant leftData = sourceModel()->data(leftIndex);
+            QVariant rightData = sourceModel()->data(rightIndex);
+
+            // make sure to compare the correct data types for sorting the current column
+            AZ_Assert(leftData.type() == rightData.type(), "EntityOutlinerSortFilterProxyModel::lessThan types do not agree!");
+            if (static_cast<QMetaType::Type>(leftData.type()) == QMetaType::QString)
+            {
+                return leftData.toString() < rightData.toString();
+            }
+            else if (static_cast<QMetaType::Type>(leftData.type()) == QMetaType::ULongLong)
+            {
+                return leftData.toULongLong() < rightData.toULongLong();
+            }
+            AZ_Error("Editor", false, "Error! Unhandled type \"%s\" in EntityOutlinerSortFilterProxyModel::lessThan", leftData.typeName());
+        }
```

Comment:
question: Is there any chance we could move this code into a helper function instead of having it duplicated (the only change is `Entity` in the assert/error calls).

Reply:
This duplication should be temporary - `EntityOutliner` in `AzToolsFramework` is the new outlienr for prefabs, I believe, so the legacy `Outliner` should go away when slices do. @AMZN-daimini can correct me if I'm wrong :)

Reply:
Yes, what Nick said. Actually, there's a heinous amount of copy-pasted code (the majority of the class) but since it's going away, there's not much point to factoring out a common base. I mean, what's the use of a common base if there's about to only be one class that derives from it? :P

Reply:
@alexmontAmazon , does it mean that the issue is fixed both in slices and prefabs? Can you confirm?

Reply:
@srikappa-amzn , yes it is fixed in both the legacy (slices) and prefab Entity Outliners.

Reply:
Ah okay got you understood, no worries then thanks for clarifying